zict/common.py

```diff
@@ -39,12 +40,12 @@ def update(*args, **kwds):
         if args:
             other = args[0]
             if isinstance(other, Mapping) or hasattr(other, "items"):
-                items += other.items()
+                items = other.items()
             else:
                 # Assuming (key, value) pairs
-                items += other
+                items = other
```

Comment:
This line is what is really different. If other is an iterator, it is not unpacked into a list; instead it is fed directly into ``_do_update`` on line 49 which in turn does not load it in memory - at least in the default implementation on line 51, which is not overridden by File.